### PR TITLE
[FLINK-19450][python][AZP] Optimize the Python CI Test

### PR DESCRIPTION
**Pull request by HuangXingBo**
## What is the purpose of the change

*This pull request will optimize the Python CI Test*

## Brief change log

  - *Optimize the pyflink test to run in latest python version if triggered by git push*

## Verifying this change

This change added tests and can be verified as follows:

  - *See the ci test whether only run in py38 currently*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by HuangXingBo** [comment]
@flinkbot run azure

**Comment by HuangXingBo** [comment]
@flinkbot run azure

**Comment by HuangXingBo** [comment]
From the running results, the python test time can be optimized from about 1 hour and 50 minutes to about 1 hour

File: tools/azure-pipelines/jobs-template.yml [hunk 1]
```diff
@@ -142,7 +142,7 @@ jobs:
       source ./tools/azure-pipelines/debug_files_utils.sh
       prepare_debug_files $(module)
       
-      ${{parameters.environment}} ./tools/ci/test_controller.sh $(module) ; TEST_EXIT_CODE=$?
+      ${{parameters.environment}} ./tools/ci/test_controller.sh $(module) $(Build.Reason); TEST_EXIT_CODE=$?
```
Comment (rmetzger) [inline tools/azure-pipelines/jobs-template.yml:145]: I don't think you need to pass the variable into the script. The variable should be available as `$BUILD_REASON` in the scripts by default.

File: tools/ci/test_controller.sh [hunk 2]
```diff
@@ -101,7 +102,7 @@ fi
 # =============================================================================
 
 if [ $STAGE == $STAGE_PYTHON ]; then
-	run_with_watchdog "./flink-python/dev/lint-python.sh" $CALLBACK_ON_TIMEOUT
+	run_with_watchdog "./flink-python/dev/lint-python.sh -c $BUILD_REASON" $CALLBACK_ON_TIMEOUT
 	EXIT_CODE=$?
```
Comment (rmetzger) [inline tools/ci/test_controller.sh:106]: these changes are also not needed

**Review by rmetzger — CHANGES_REQUESTED** [review]
I believe most of the changes in this PR are not needed, because you can access `$BUILD_REASON` directly from the environment.

**Comment by HuangXingBo** [comment]
@rmetzger Thanks a lot for the review. I have removed the content of passing the variable of BUILD_REASON.

**Review by dianfu — APPROVED** [review]
@HuangXingBo Thanks a lot for the PR. Good work! It reduces the execution time of python tests a lot. The PR LGTM to me. +1 from my side. 
@rmetzger Thanks a lot for the review! Would you like to take a further look at of this PR?

**Review by rmetzger — APPROVED** [review]
Change looks good

**Comment by dianfu** [comment]
@rmetzger Thanks for the final review. Merging...

**Pull request closed without merging**